lib/user.js

```diff
@@ -52,6 +76,46 @@ class UserModel extends BaseModel {
     unsealToken(sealed, callback) {
         return iron.unseal(sealed, this.password, iron.defaults, callback);
     }
+
+    /**
+     * Get permissions on a specific repo
+     * @param  {String}  config.username        Username
+     * @param  {String}  config.scmUrl          The scmUrl of the repository
+     * @return {Function} callback              fn(err, permissions) where permissions is an object
+     *                                          that contains the permissions for [admin, push, pull]
+     *                                          Example: {admin: false, push: true, pull: true}
+     */
+    getPermissions(config, callback) {
+        const id = this.generateId({ username: config.username });
+
+        async.waterfall([
+            (next) => {
+                this.get(id, next);
+            },
+            (user, next) => {
+                this.unsealToken(user.token, next);
+            },
+            (unsealed, next) => {
+                const matched = (schema.config.regex.SCM_URL).exec(config.scmUrl);
```

Comment:
Does this count?
https://github.com/screwdriver-cd/screwdriver/blob/f4000828656687d784b21afbb51b6dd87c427a22/plugins/pipelines/create.js#L73

The scmUrl is passed in is from the payload, which should be validated in the api already. I can still add a check here if you prefer but I think it will never happen.